components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -402,35 +405,58 @@ public void updateBusinessOwner(BusinessOwner businessOwner) throws AppManagemen
      * Save a business owner.
      * @param businessOwner
      */
-    public void saveBusinessOwner(BusinessOwner businessOwner){
+    public int saveBusinessOwner(BusinessOwner businessOwner) throws AppManagementException {
 
         Connection connection = null;
         PreparedStatement statementToInserBusinessOwner = null;
         PreparedStatement statementToInsertBusinessOwnerDetails = null;
+        String businessOwnerName = "";
+        int businessOwnerId = 0;
         try {
+            businessOwnerName = businessOwner.getBusinessOwnerName();
             if (log.isDebugEnabled()) {
-                log.debug("Adding a Business Owner" + businessOwner.getBusinessOwnerName());
+                log.debug("Adding a Business Owner" + businessOwnerName);
             }
             connection = APIMgtDBUtil.getConnection();
             String queryToInsertRecord = "INSERT INTO BUSINESS_OWNER(OWNER_NAME,OWNER_EMAIL,OWNER_DESC,OWNER_SITE)"
                     + " VALUES (?,?,?,?)";
 
-            statementToInserBusinessOwner = connection.prepareStatement(queryToInsertRecord);
-            statementToInserBusinessOwner.setString(1, businessOwner.getBusinessOwnerName());
+            statementToInserBusinessOwner = connection.prepareStatement(queryToInsertRecord,
+                                                                      Statement.RETURN_GENERATED_KEYS);
+            statementToInserBusinessOwner.setString(1, businessOwnerName);
             statementToInserBusinessOwner.setString(2, businessOwner.getBusinessOwnerEmail());
             statementToInserBusinessOwner.setString(3, businessOwner.getBusinessOwnerDescription());
             statementToInserBusinessOwner.setString(4, businessOwner.getBusinessOwnerSite());
-            statementToInserBusinessOwner.executeUpdate();
+            int affectedRows = statementToInserBusinessOwner.executeUpdate();
+
+           if (affectedRows == 0) {
+                throw new AppManagementException("Saving business owner user : " + businessOwnerName + " is failed, no "
+                                                 + "rows affected.");
+            }
+            ResultSet generatedKeys = statementToInserBusinessOwner.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                businessOwnerId = generatedKeys.getInt(1);
+            }
+            else {
+                throw new SQLException("Creating user failed, no ID obtained.");
+                   }
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/4642888b8ad6934dcda162142607b8549b0728e5